tests/cases/assetstore_test.py

```diff
@@ -22,9 +22,12 @@
 from girder.utility.progress import ProgressContext
 from girder.utility.s3_assetstore_adapter import makeBotoConnectParams
 from girder.utility import path as path_util
+from girder_worker.app import app
 
 
 def setUpModule():
+    app.conf.task_always_eager = True
+    app.conf.task_eager_propagates = True
```

Comment:
These are the two config options that will need to be used in testing to have tasks run eagerly. If set, we shouldn't need to change (much) testing code, as it should behave the same as if it was running the code in the request thread.